fetool/pages/_app.js

```diff
@@ -1,7 +1,46 @@
-import '../styles/globals.css'
+import "../styles/globals.css";
+import { ToastContainer,toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+import { parseCookies  } from 'nookies'
 
 function MyApp({ Component, pageProps }) {
-  return <Component {...pageProps} />
+  return (
+    <>
+      <ToastContainer autoClose={1000}/>
+      <Component {...pageProps} />
+    </>
+  );
 }
 
-export default MyApp
+function redirectUser(ctx, location) {
+  if (ctx.req) {
+    //302-Resource requested has been temporarily moved to the URL given by the Location header.
+      ctx.res.writeHead(302, { Location: location });
+      ctx.res.end();
+  } else {
+      Router.push(location);
+  }
+}
+
+MyApp.getInitialProps = async ({Component, ctx}) => {
+  let pageProps = {}
+  //Accessing cookies in NextJS from server side
+  const jwt = parseCookies(ctx).jwt
+
+  if (Component.getInitialProps) {
+      pageProps = await Component.getInitialProps(ctx)
+  }
+
+  if (!jwt) {
+      if (ctx.pathname === "/clients") {
+        redirectUser(ctx, "/");
+      }
+  }
+
+  return {
+      pageProps,
+     
```

Comment:
Use prettier extension to avoid empty lines in between code. Ensure that the settings that we configured are in tact.

Reply:
It is done